### PR TITLE
Ds session 5 assignment

### PR DESCRIPTION
**Pull request by meta-jugal-kukreja**

### CODE REVIEW

File: DS-Session-5-Assignment/src/question2/EmployeeList.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	 * This method sorts the given employee list with the insertion sort method.
+	 */
+	void sortEmployeeList()
+	{
+		Node currentNode = linkedListHead;
+		while(currentNode != null)
+		{
+			Node nextNode = currentNode.next;
+			if((sortedLinkedListHead == null) || 
+					(sortedLinkedListHead.getEmployeeSalaray() <= currentNode.getEmployeeSalaray()))
```
Comment (meta-deepak-goyal) [inline DS-Session-5-Assignment/src/question2/EmployeeList.java]: Need an explanation of this sorting.

File: DS-Session-5-Assignment/src/question1/BinarySearchTree.java [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+	
+	 Node root;
+	
+	/**
+	 * Constructor to add data from the JSON File
+	 * @param object is the whole JSON object
+	 */
+	public BinarySearchTree(JSONObject object) {
+		this.root = null;
+        for (Object key : object.keySet()) {
```
[the remark this answers is not in the document]
Comment (meta-jugal-kukreja) [reply DS-Session-5-Assignment/src/question1/BinarySearchTree.java]: Done

File: DS-Session-5-Assignment/src/question1/BinarySearchTree.java [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+     * This method adds node to a tree by calling another method which inserts the node recursively in the tree
+     * @param key is the key
+     * @param value is the value of the key
+     * @return true if tree node is added successfully else false
+     */
+	@Override
+	public void addKeyValuePair(String key, String value) {
+		
+		 // if key is empty
+        if (key == null || key.equals(" ")) {
```
[the remark this answers is not in the document]
Comment (meta-jugal-kukreja) [reply DS-Session-5-Assignment/src/question1/BinarySearchTree.java]: Done

File: DS-Session-5-Assignment/src/question1/BinarySearchTree.java [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+
+	/**
+     * This method deletes a node with a given key
+     * @param key is the key to be deleted
+     */
+	@Override
+	public void deleteKeyValuePair(String key) {
+		// if key is empty
+        if (key == null || key.equals(" ")) {
+            throw new AssertionError("invalid key");
```
[the remark this answers is not in the document]
Comment (meta-jugal-kukreja) [reply DS-Session-5-Assignment/src/question1/BinarySearchTree.java]: Done

File: DS-Session-5-Assignment/src/question1/BinarySearchTree.java [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+		else
+		{
+			// node with only one child or no child
+            if (currentNode.leftChild == null)
+                return currentNode.rightChild;
+            else if (currentNode.rightChild == null)
+                return currentNode.leftChild;
+
+            //Else find the minimum in the right sub tree and replace the current node with it and delete that
+            Data data = minValue(currentNode.rightChild);
```
[the remark this answers is not in the document]
Comment (meta-jugal-kukreja) [reply DS-Session-5-Assignment/src/question1/BinarySearchTree.java]: ok

File: DS-Session-5-Assignment/src/question2/EmployeeList.java (hunk 1 above)
Comment (meta-jugal-kukreja) [reply DS-Session-5-Assignment/src/question2/EmployeeList.java]: ok

**Pull request merged**